src/sentry/shared_integrations/client.py

```diff
@@ -44,9 +44,11 @@ def rel(self):
 
     @classmethod
     def from_response(self, response, allow_text=False):
+        if response.request.method == "HEAD":
+            return TextApiResponse(response.status_code, response.headers, response.status_code)
```

Comment:
@ceorourke why are we passing the status code as the `text` field? I think we should just return an instance of the `BaseApiResponse` since HEAD requests shouldn't have any information in the response other than headers and a status code.

Reply:
that makes sense